core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -109,6 +113,95 @@ public DefaultRequestLog(RequestContext ctx) {
         this.ctx = requireNonNull(ctx, "ctx");
     }
 
+    @Override
+    public void addChild(RequestLog child) {
+        checkState(!hasLastChild, "last child is already added");
+        requireNonNull(child, "child");
+        if (children == null) {
+            // first child's all request-side logging events are propagated immediately to the parent
+            children = new ArrayList<>();
+            propagateRequestSideLog(child);
+        }
+        children.add(child);
+    }
+
+    private void propagateRequestSideLog(RequestLog child) {
+        child.addListener(log -> startRequest(log.channel(), log.sessionProtocol(), log.host()), REQUEST_START);
+        child.addListener(log -> serializationFormat(log.serializationFormat()), SCHEME);
+        child.addListener(log -> requestHeaders(log.requestHeaders()), REQUEST_HEADERS);
+        child.addListener(log -> requestContent(log.requestContent(), log.rawRequestContent()),
+                          REQUEST_CONTENT);
+        child.addListener(log -> {
+            if (log.requestCause() != null) {
+                endRequest(log.requestCause());
+            } else {
+                endRequest();
```

Comment:
Ditto - Could we set the end time explicitly so that the end time of the parent request log is same with that of the first child? You could add a private variant of `endRequest()` which accepts the start time as a parameter.